docs/docs/adrs/adr-015-partial-set-security.md

```diff
@@ -0,0 +1,308 @@
+Then, because we do not use the canonical fee pool and hence rewards are not automatically distributed, we will create a new and modified version of [AllocateTokens](https://github.com/cosmos/cosmos-sdk/blob/v0.47.7/x/distribution/keeper/allocation.go#L14) as follows:
+
+```
+func (k Keeper) AllocateTokensFromPool(ctx sdk.Context, poolAddress string, totalPreviousPower int64, bondedVotes []abci.VoteInfo) {
+	rewardsCollectedInt := k.bankKeeper.GetAllBalances(ctx, poolAddress)
+	// rest in the same that `AllocateTokens` operates
+	...
+```
+
+In [TransferRewardsToFeeCollector](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/provider/keeper/distribution.go#L61) we then do:
+
+```
+for _, chain := range consumerChains {
+    ...
+    totalPower := 0
+    var []abci.VoteInfo votes
+    for _, val := range (validators that opted in or Top N) {
+        if val has opted in for more than X blocks {
+            votes = append(votes, []abci.VoteInfo{{Validator: val, SignedLastBlock: true}})
+        }
+    }
+    
+    k.AllocateTokens(ctx, pool associated with this chain, totalPower, votes)
+    ...
+}
+```
+
+instead of calling:
+```
+// 4. Transfer the balance to the fee collector address
+err := k.bankKeeper.SendCoinsFromModuleToModule(
+    ctx,
+    types.ConsumerRewardsPool,
+    k.feeCollectorName,
+    sdk.NewCoins(balance),
+)
+```
+
+Note that we would only distribute rewards to validators that are opted in but are opted in for some time (e.g., 10000 blocks) to avoid cases where validators opt in and out in short intervals just in order to receive rewards.
+
+### Misbehaviour
+
+#### Fraud votes
+For fraud votes, we add a new type `FraudVoteProposal` similar to [EquivocationProposal](https://github.com/cosmos/interchain-security/pull/703). The time to detect incorrect execution evidence is 7 days (the same time as that of detecting a light client attack on a client with a `trustingPeriod` of 14 days); We have 7 days to detect an incorrect execution and then 14 days for the proposal to be accepted.
+
+##### When are fraud votes eligible?
+Because fraud votes allow the slashing of a validator through a governance proposal, we are cautious on when a fraud vote is eligible and try to restrict their eligibility as much as possible to avoid an adversary proposing a bogus fraud vote.
+As described earlier, because in Top N chains we consider that `N > 50%` we only consider validators that have opted in an Opt In chain. Fraud votes are also **only** intended to be used for **malicious** behaviour, for example, in a scenario similar to what [happened to Neutron](https://blog.neutron.org/neutron-halt-post-mortem-927dbe4540c8), validators should vote against slashing those validators.
+
+#### Message
+```
+message FraudVoteProposal {
```

Comment:
Is there a reason why this proposal is just for one validator fraud vote?

Reply:
You mean in contrast to being able to slash multiple validators for the same fraud vote?
This could be done as well, similarly to [EquivocationProposal](https://github.com/cosmos/interchain-security/pull/703). Nevertheless, having multiple validators to be slashed on a proposal could lead to the following confusing scenario: A proposal is proposed with validators `V1, V2, V3` but a validator `V` only considers that `V1, V2` performed incorrect execution but not `V3`. What should `V` vote for ? Yes or No?
It seems better to only consider one validator per proposal to avoid such edge cases. Of course, in this case we might have multiple proposals but there's i) no cost involved in this because the proposal deposit would be returned at the end, and ii) if someone has created one proposal already it would be straightforward to create another one with a different validator address.

Reply:
Resolving this because we will not talk about fraud votes in this ADR anymore.